lib/pure/typetraits.nim

```diff
@@ -107,6 +107,10 @@ since (1, 3, 5):
 
 import std/macros
 
+macro elementsCount*(Enum: type enum): int =
+  ## Returns the number of items in the enumeration `Enum`.
+  newIntLitNode(Enum.getImpl[2].len - 1)
+
```

Comment:
this fails:
```nim
when true:
  import macros
  macro len*(T: typedesc[enum]): int =
    ## Returns the number of items in the enumeration `T`.
    newLit(T.getImpl[2].len - 1)

  type Foo* = enum k0 k1
  static:
    doAssert Foo.len == 2
    type Foo2 = Foo
    doAssert Foo2.len == 2
```

the fix is easy, look at the the other procs in this module

Reply:
Fixed.